### DISCUSSION TITLE
Getting a spawn UNKNOWN error? Tried lots of things. No idea what direction to go.

### DISCUSSION THREAD

**Discussion by jcolley** [jef/streetmerchant#1192] [Help]
I receive the below error and cannot seem to get this wonderful utility to start.

What I've tried:
- Running a fresh clone of the latest release in Windows CMD shell, with and without Administrator access
- Running a fresh clone of the latest release in WSLv1 Ubuntu 18.04 LTS
- Removing all unused .env variables
- Following the Windows installation guide
- Following the Linux installation guide
- Searching for similar issues
- Reinstalling NPM (both OS's)
- Restarting PC (several times throughout)

OS: Win10 20H2 / OS: Ubuntu 18.04 LTS (WSL)

Here's the `npm run start` output:
```
[3:32:47 AM] info :: ℹ selected stores: amazon, amd, bandh, bestbuy, expert, gamestop, microcenter, newegg, officedepot, target, walmart
[3:32:47 AM] info :: ℹ selected brands: evga, nvidia, amd
[3:32:47 AM] info :: ℹ selected models: ftw3, ftw3 ultra, 5900x, 5950x, founders edition
[3:32:47 AM] info :: ℹ selected series: 3060ti, 3070, 3080, 3090, rx6800, rx6800xt, rx6900xt, ryzen5600, ryzen5800, ryzen5900, ryzen5950, sonyps5c, sonyps5de, xboxss, xboxsx
[3:32:48 AM] error :: ✖ something bad happened, resetting streetmerchant in 5 seconds spawn UNKNOWN {
  "errno": "UNKNOWN",
  "code": "UNKNOWN",
  "syscall": "spawn",
  "stack": "Error: spawn UNKNOWN\n    at ChildProcess.spawn (internal/child_process.js:407:11)\n    at Object.spawn (child_process.js:548:9)\n    at BrowserRunner.start (C:\\Users\\name\\Projects\\streetmerchant\\node_modules\\puppeteer-extra\\node_modules\\puppeteer\\lib\\Launcher.js:77:30)\n    at ChromeLauncher.launch (C:\\Users\\name\\Projects\\streetmerchant\\node_modules\\puppeteer-extra\\node_modules\\puppeteer\\lib\\Launcher.js:242:12)\n    at async PuppeteerExtra.launch (C:\\Users\\name\\Projects\\streetmerchant\\node_modules\\puppeteer-extra\\dist\\index.cjs.js:129:25)\n    at async main (C:\\Users\\name\\Projects\\streetmerchant\\build\\index.js:36:15)\n    at async loopMain (C:\\Users\\name\\Projects\\streetmerchant\\build\\index.js:72:9)"
```

Here's a copy of my current `.env`:
```
AUTO_ADD_TO_CART="true"
DESKTOP_NOTIFICATIONS="true"
LOW_BANDWIDTH="true"
MAX_PRICE_SERIES_3060TI="450"
MAX_PRICE_SERIES_3070="599"
MAX_PRICE_SERIES_3080="799"
MAX_PRICE_SERIES_3090="1000"
MAX_PRICE_SERIES_CORSAIR_SF=""
MAX_PRICE_SERIES_RX6800="500"
MAX_PRICE_SERIES_RX6800XT="500"
MAX_PRICE_SERIES_RX6900XT="500"
MAX_PRICE_SERIES_RYZEN5600="500"
MAX_PRICE_SERIES_RYZEN5800="500"
MAX_PRICE_SERIES_RYZEN5900="500"
MAX_PRICE_SERIES_RYZEN5950="500"
MICROCENTER_LOCATION="parkville,rockville"
NVIDIA_ADD_TO_CART_ATTEMPTS="3"
SHOW_ONLY_BRANDS="evga,nvidia,amd"
SHOW_ONLY_MODELS="ftw3,ftw3 ultra,5900x,5950x,founders edition"
STORES="amazon,amd,bandh,bestbuy,expert,gamestop,microcenter,newegg,officedepot,target,walmart"
```

Thanks for any insight or suggestions.

**Comment by jef** [comment]
First, I would remove `NVIDIA_ADD_TO_CART_ATTEMPTS="3"`, as it's only for the `nvidia` and `nvidia-api` stores.

Second, I would move your `.env` to a backup folder and try to run the app (no configuration). If it works, then it's your config. If it doesn't then you know it is your config!

Let me know how it works from there.

Thanks!

**Comment by jcolley** [comment]
Thanks for the help.
I decided to start from scratch. Deleted the existing `streetmerchant` directory and did a `git clone` to grab a fresh copy.
Performed `npm install` and `npm install @types/express @types/qs@6.9.4` to ensure dependencies are met.

After running `npm run start` - without creating a `.env` or changing any files, I still get the same `something bad happened...` and `spawn UNKNOWN...` error.

Since there is no config. It can't be my config, no?

**Comment by jef** [comment]
> `npm install @types/express @types/qs@6.9.4`

I don't know if this is necessary. Where did you see this was the case?

I see now that you're using WSL. I'm not 100% on the dependencies needed, but do you have these dependencies in this list?

https://github.com/jef/streetmerchant/wiki/Installation:-Linux:-Debian-based

**Comment by jcolley** [comment]
I want to be very clear: I've tried this separately, with a fresh clone each time, in Win10's `cmd` shell, as well as WSLv1 Ubuntu 18.04 LTS, and received the same error in both environments. Because of this, I am now focusing on Win10's `cmd` shell.

The first time I ran `npm install` in each environment, I received warnings that `@types/express` and `@types/qs@6.9.4` must be installed manually. They are typescript dependencies and after running `npm install @types/express @types/qs@6.9.4`, the warning went away and I assume all is well on the dependency front.

I did manage to get it working just now. Having been a sys admin for over 20 years, I incorrectly assumed that I could loosely follow the step-by-step Windows installation [instructions](https://github.com/jef/streetmerchant/wiki/Installation:-Windows), and (as my original post shows) I was cloning the project into `C:\Users\name\Projects\streetmerchant\` as that's where I wanted the application to live.
On a hunch, I tried @ `C:\streetmerchant\` as the instructions clearly state, and it works out-of-the-box.

From my experience as a Windows admin, it's not a great idea to start adding applications and directories at the root `C:\` directory level. After all, Microsoft themselves separate `Program Files` from `User` files there for a reason and at some point (at least in the past) it was even against the Windows software design guides as published by Microsoft.

As long as we know what we're doing, it won't really hurt anything, but that doesn't make it right.
Can you share with me the reason the application requires the path: `C:\streetmerchant\` and fails inside a user directory? I am guessing it has to do with either the $PATH variable or Window's limits on long file paths, but I'd like to investigate making this work elsewhere if you could nudge me in the right direction by sharing your reasoning.
I'll be happy to make a PR with any required changes that would enable this to work.

**Reply by jef** [reply]
That's interesting, because I've put streetmerchant in several different places on my Windows machine while testing. I honestly can't duplicate what you've ran into unforunately.

Perhaps the system environment for node location is off, but I never had to touch that on my Windows machine. I don't have _any_ development work on there really, so it's a good place for me to test things.
